### PR TITLE
Cache keyPair for each Node

### PR DESCRIPTION
**Pull request by Mykhailenko**
In order to improve Task Dispatching time, we decided to have single key pair for each Node and not for each task. Thus they are cached in RMNodeStarter.

### CODE REVIEW

File: rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java [hunk 1]
```diff
@@ -111,6 +111,32 @@
 
     protected String nodeSourceName = null;
 
+    public static KeyPair getKeyPairForNode() throws NoSuchAlgorithmException, NodeException {
```
Comment (sonarqube-activeeon) [inline rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java]: MINOR Remove the declaration of thrown exception 'java.security.NoSuchAlgorithmException', as it cannot be thrown from method's body. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [TaskLauncher.java#L31](https://github.com/ow2-proactive/scheduling/blob/b81c4f5f9350e5fa183fce3d5ad9dc5ce25deac7/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/TaskLauncher.java#L31): Remove this unused import 'java.security.KeyPairGenerator'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TaskLauncher.java#L34](https://github.com/ow2-proactive/scheduling/blob/b81c4f5f9350e5fa183fce3d5ad9dc5ce25deac7/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/TaskLauncher.java#L34): Remove this unused import 'java.security.SecureRandom'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/TaskLauncher.java [hunk 2]
```diff
@@ -449,11 +453,33 @@ public void getStoredLogs(AppenderProvider logSink) {
...
+     * because in some test (e.g. TaskLauncherTest) we do not mock PAActiveObject
+     * so it is easy to mock this method.
+     * @return
+     * @throws NodeException
+     */
+    public String nodeUrl() throws NodeException {
+        return PAActiveObject.getNode().getNodeInformation().getURL();
+    }
+
+    private static KeyPair getKeyPair(String nodeUrl) throws NodeException, NoSuchAlgorithmException {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/TaskLauncher.java]: MINOR Remove the declaration of thrown exception 'org.objectweb.proactive.core.node.NodeException', as it cannot be thrown from method's body. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (fviale) [reply rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java]: After a second thought, I think that creating a public static method in RMNodeStarter returning a KeyPair is not secure enough.

**Pull request closed without merging**